src/main/java/ch/njol/skript/events/EvtEntityPotion.java

```diff
@@ -0,0 +1,93 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser;
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.EntityPotionEffectEvent;
+import org.bukkit.potion.PotionEffectType;
+
+import javax.annotation.Nullable;
+
+public class EvtEntityPotion extends SkriptEvent {
+
+	static {
+		Skript.registerEvent("Entity Potion Effect", EvtEntityPotion.class, EntityPotionEffectEvent.class,
+				"add[ing] [of] [entity] potion effect [[of] %potioneffecttypes%]",
```

Comment:
I am not a big fan of these syntaxes, maybe we can come up with better ones. Something like `on entity potion effect modification of %potioneffecttypes%`,

since we will already need an expression for the possible actions so that users can check for in a single event e.g. `if [modification] action is adding`.

I even think if we create a general expression for `action` that can be used in many different places and would be parsed correctly based on the event.

Reply:
Would you like me to create this expression?

Reply:
> I even think if we create a general expression for action that can be used in many different places and would be parsed correctly based on the event.

This is out-of-scope for this PR and will require a decent work therefore you can just add the action for this event as a simple expression

Reply:
Been trying to do this for like 2-3 days now, since you mentioned it basically. But I just can't get it to work, for whatever reason. So I think i'm going to compromise, and just make it "on entity potion effect modification of %potioneffecttypes%", like you said,  and then whenever you guys release your action expression, this event can be updated.

Reply:
No worries, I will try to suggest the code by tomorrow